pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+        """Set the quadratic attenuation constants."""
+        self.SetAttenuationValues(values)
+
+    # TODO: implement transformation_matrix here?
+
+    @property
+    def light_type(self):
+        """Return the light type.
+
+        The default light type is a scene light which lives in world
+        coordinate space.
+
+        A headlight is attached to the camera and always points at the
+        camera's focal point.
+
+        A camera light also moves with the camera, but it can have an
+        arbitrary relative position to the camera. Camera lights are
+        defined in a coordinate space where the camera is located at
+        (0, 0, 1), looking towards (0, 0, 0) at a distance of 1, with
+        up being (0, 1, 0). Camera lights use the transform matrix to
+        establish this space.
+
+        The property returns class constant values from an enum:
+            - Light.HEADLIGHT == 1
+            - Light.CAMERA_LIGHT == 2
+            - Light.SCENE_LIGHT == 3
+
+        """
+        return LightType(self.GetLightType())
+
+    @light_type.setter
+    def light_type(self, ltype):
+        """Set the light type.
+
+        Either an integer code or a class constant enum value must be used.
+
+        """
+        if not isinstance(ltype, int):
+            # note that LightType is an int subclass
+            raise TypeError('Light type must be an integer subclass,'
+                            f' got {ltype} instead.')
+        self.SetLightType(ltype)
+
+    @property
+    def is_headlight(self):
+        """Return whether the light is a headlight."""
+        return bool(self.LightTypeIsHeadlight())
+
+    @property
+    def is_camera_light(self):
+        """Return whether the light is a camera light."""
+        return bool(self.LightTypeIsCameraLight())
+
+    @property
+    def is_scene_light(self):
+        """Return whether the light is a scene light."""
+        return bool(self.LightTypeIsSceneLight())
+
+    @property
+    def shadow_attenuation(self):
+        """Return the shadow intensity.
+
+        By default a light will be completely blocked when in shadow.
+        By setting this value to less than 1 you can control how much
+        light is attenuated when in shadow.
+
+        """
+        return self.GetShadowAttenuation()
+
+    @shadow_attenuation.setter
+    def shadow_attenuation(self, shadow_intensity):
+        """Set the shadow intensity."""
+        self.SetShadowAttenuation(shadow_intensity)
+
+    #### Everything else ####
+
+    def set_color(self, color):
+        # TODO: do we want to remove this?
+        # TODO: do we want instead to set _all_ lights in pyvista?
+        # TODO: do we want to support a three-arg form?
+        """Set the diffuse and specular color of the light.
+
+        Parameters
+        ----------
+        color : tuple(float)
+            The color that should be set for diffuse and specular.
+
+        """
+        self.SetColor(color)
```

Comment:
```suggestion
```

Completely remove it.  We're creating a new API via this class, and it's only kept within VTK for backwards compatibility, and will still be there with the C++ API.